abstractions/python/kiota/abstractions/serialization/parse_node_proxy_factory.py

```diff
@@ -0,0 +1,42 @@
+from io import BytesIO
+from typing import Callable
+
+from .parsable import Parsable
+from .parse_node import ParseNode
+from .parse_node_factory import ParseNodeFactory
+
+
+class ParseNodeProxyFactory(ParseNodeFactory):
+    """Proxy factory that allows the composition of before and after callbacks on existing factories
+    """
+    def __init__(
+        self, concrete: ParseNodeFactory, on_before: Callable[[Parsable], None],
+        on_after: Callable[[Parsable], None]
+    ) -> None:
+        """Creates a new proxy factory that wraps the specified concrete factory while composing
+        the before and after callbacks.
+
+        Args:
+            concrete (ParseNodeFactory): The concrete factory to wrap.
+            on_before (Callable[[Parsable], None]): The callback to invoke before the
+            deserialization
+            of any model object.
+            on_after (Callable[[Parsable], None]): The callback to invoke after the deserialization
+            of any model object.
+        """
+        self._concrete = concrete
+        self._on_before = on_before
```

Comment:
this should probably wrapped in a callback instead of replacing the existing so that both callbacks (the constructor passed and the concrete) are called

Reply:
This is just Python's syntax for initializing/assigning values to the data members of the class when an object of class is created. 

Also updated the logic to make sure all callbacks are called. https://github.com/microsoft/kiota/pull/1058/commits/81a71121dbfab58997dab82ecd5a3607b7e59fca